desktop/libs/notebook/src/notebook/connectors/spark_shell.py

```diff
@@ -151,21 +151,32 @@ class SparkApi(Api):
   STANDALONE_JOB_RE = re.compile("Got job (\d+)")
 
   @staticmethod
-  def get_properties():
-    return SparkConfiguration.PROPERTIES
+  def get_props(properties=None):
+    props = dict([(p['name'], p['value']) for p in SparkConfiguration.PROPERTIES])
+    if properties is not None:
+      props.update(dict([(p['name'], p['value']) for p in properties]))
 
-  def create_session(self, lang='scala', properties=None):
-    if not properties:
-      config = None
-      if USE_DEFAULT_CONFIGURATION.get():
-        config = DefaultConfiguration.objects.get_configuration_for_user(app='spark', user=self.user)
+    return props
 
-      if config is not None:
-        properties = config.properties_list
-      else:
-        properties = self.get_properties()
+  @staticmethod
+  def get_properties(props=None):
+    properties = list()
+    for p in SparkConfiguration.PROPERTIES:
+      properties.append(p.copy())
+
+    if props is not None:
+      for p in properties:
+        if p['name'] in props:
+          p['value'] = props[p['name']]
 
-    props = dict([(p['name'], p['value']) for p in properties]) if properties is not None else {}
+    return properties
+
+  def create_session(self, lang='scala', properties=None):
+    if not properties and USE_DEFAULT_CONFIGURATION.get():
+      user_config = DefaultConfiguration.objects.get_configuration_for_user(app='spark', user=self.user)
+      properties = user_config.properties_list
```

Comment:
And also move this property logic into the get_properties()?

Reply:
Method `get_props`-  generate dictionary in format required by livy.
Method `get_properties(props)` (renamed to `to_properties(props)`) - convert props back to list with all metadata like nice_name,help_text etc.
Method get_properties used only there:
```
    return {
        'type': lang,
        'id': response['id'],
        'properties': self.to_properties(props)
    }
```